src/index.js

```diff
@@ -29,36 +26,56 @@ class Resource {
    * {
    *   id: '5869485767494'
    *   params: {
-   *     limit: 5
+   *     limit: 5,
+   *     since: 1467964293258
    *   }
    * }
    *
    * @param query
    * @returns {Promise}
    */
-  get (query) {
-    if (!query) {
-      query = {};
-    }
+
+  get(query, results) {
+    var query = query || {}
+    var results = results || [];
+    var that = this;
+    var deferred = Promise.defer();
 
     this.signatureFactory.setUrl(this.url);
     this.signatureFactory.handleQuery(query);
     const signature = this.signatureFactory.sign();
 
-    const options = {
-      url: this.getBaseUrl() + signature.url,
+    var requestOptions = {
+      protocol: 'https:',
+      host: this.config.host,
+      path: signature.url,
       headers: signature.headers
     };
 
-    return new Promise((resolve, reject) => {
-      request(options, (error, response, body) => {
-        if (!error && response.statusCode == 200) {
-          resolve(JSON.parse(body));
+    https.get(requestOptions,function(res) {
+      var str = '';
+      var answer = {};
+
+      res.on('data',function(chunk) {
+        str += chunk;
+      });
+
+      res.on('end',function() {
+        answer = JSON.parse(str);
+        results = results.concat(answer.items);
+
+        if (answer.hasMore && that.config.iterator) {
+          query = assign(query, {params: { since: answer.lastTimestamp }});
+          that.get(query, results).then((results) => {
+            deferred.resolve(results);
+          });
         } else {
-          reject(body);
+          deferred.resolve(results);
         }
       });
     });
```

Comment:
You could handle the error here as well and reject promise. Take a look [here](https://nodejs.org/api/https.html#https_https_get_options_callback).

```
.on('error', (e) => {
  deferred.reject(e); // Or a string representation of this error
});
```